Updated to use OAuth 2.0

Purzifal opened the pull request:
Since you guys are taking some time to update python stuff here is a pull request to fix the chat example

- Updated to use OAuth 2.0
- Added run.bat which fixes crash when certain characters are printed to the console (Windows Only)
- removed the sending 'Hi!' every second
- Added !ping chat command
- Added some chat formatting (can get raw output by commenting 1 line)
- Tested on 3 OS platforms, 2 python versions (see readme.md)

- Updated readme.md to be more accurate

Commit:
Updated for Auth 2.0 authentication

Updated to use OAuth 2.0 due to previous method being depreciated.
Update readme.md to reflect changes and instructions for Windos and
Linux operating systems. added !ping command and message formating (raw
format available with uncommenting code)

Commit:
updated

Commit:
make flake8 happy

Commit:
Fixed docstring description

fixed docstring description for method

Commit:
Multiple changes, couple addidtions

changed the way selecting the chat server selection is done. Sent bot
"system" messages though emit instead of print as it allows for handling
data better for more advanced bots. Added comments where possible. Added
couple variables to config.py for ChatEventHandler.py. Added
ChatEventHandler to format the text output in a readable format. Moved a
couple functions from example.py to ChatEventHandler.py

2Cubed reviewed (approved):
Looks fantastic. Very well done.

ProbablePrime commented:
@connor4312 As you were the original creator of this would you mind giving this a review when you get the chance. Thanks!

Innectic commented on ChatEventHandler.py:
```diff
@@ -0,0 +1,126 @@
+        print('CHAT EVENT HANDLER: ' + str(data))
+
+
+def typereply(data):
```
Making these `type` methods more "pythonic" could be a thing... `typereply` -> `type_reply`

Innectic commented on ChatEventHandler.py:
```diff
@@ -0,0 +1,126 @@
+    msg = ''
+
+    # gets the message list amount
+    list_count = len(data['message']['message'])
```
You could use `join` here to make a lot less code. Something like the following should work:

```python
msg = ' '.join(item["text"] for item in data["message"]["message"])
```

Innectic commented on setup.py:
```diff
@@ -1,3 +1,6 @@
+'''Sets up required dependancies to run the bot'''
```
s/dependancies/dependencies

Innectic reviewed (changes requested):
Not terrible, but could use a few changes, IMO

Commit:
made chagnes suggested by Innectic

fixed spelling mistake in setup.py , Changed code for format_msg
function and renamed typereply to type_reply in ChatEventHandler.py

capgames commented:
@Innectic thanks for the suggestions, The ChatEventHandler.py is my little addition just to format the chat events into a more readable format. I know the code could be better and is something I am working on for my own project 👍

connor4312 commented:
Hey, sorry for the delay here. I'll make a note to myself to get this reviewed tomorrow.

Innectic commented on ChatEventHandler.py:
```diff
@@ -112,15 +111,7 @@ def typemethod(data):
-    for counter in range(0, list_count):
-        # joins the message together
-        msg += str(data['message']['message'][counter]['text'])
+    msg = ''.join(item["text"] for item in data["message"]["message"])
```
*Almost* perfect. Add a space between the quotes. `.join` joins all the items in a list by the delimiter, which in this case, is the text between the `''` :D

Purzifal replied:
when i tested this with and without the space. 

* With the space:  between quotes it added an extra space between the joins
* Without the space: no extra space between the joins

Sere is the results with having the space for that line (top) and without the space (bottom)
![image](https://cloud.githubusercontent.com/assets/20742198/25607343/f6403884-2f0d-11e7-88e5-ab2f607f68e3.png)

This is why I removed the space to make the text formatting look better. the joins are before and after the emoji and @ mention

I can add the space, however, it make the formatting look off

Innectic replied:
Huh, I didn't think the beam packet included the space

connor4312 commented on ChatEventHandler.py:
```diff
@@ -0,0 +1,117 @@
+    elif data['event'] == 'PollEnd':
+        print('The poll started by {} has ended'.format(
+            data['data']['author']['user_name']))
+        CONFIG.POLL_SPAM = False
```
It could be cleaner to have a dict of event names to functions that are executed, then call `those_funcs[data['event']](data)`

connor4312 commented on README.md:
```diff
@@ -4,10 +4,21 @@ This is a basic chat bot framework for Beam, built to serve as a basic example o
-- run the example chatbot (`python example.py`)
+* install dependencies
+  * Windows : `python setup.py install`
+  * Linux   : `sudo python setup.py install`
```
This is not necessary if you use a virtualenv (generally best practice) or use a non-privileged install (`/usr/local/...`). I'd get rid of this.

connor4312 commented on ChatEventHandler.py:
```diff
@@ -0,0 +1,117 @@
+    msg = ''.join(item["text"] for item in data["message"]["message"])
+
+    # returns the message
+    return msg
```
If splitting into multiple files, this should be moved to an `examples` folder.

connor4312 commented on config.example.py:
```diff
@@ -1,11 +1,25 @@
 
-# The channel ID you want to log into
-CHANNEL = 1
+# This is up to you to obtain
```
Instructions?

connor4312 commented on config.example.py:
```diff
@@ -1,11 +1,25 @@
+# enables/disables raw chat json
+CHATDEBUG = False
+
+# enables/disables spam that is caused by polls
```
Details?

connor4312 commented on example.py:
```diff
@@ -1,29 +1,26 @@
 
 if __name__ == "__main__":
-    chat = create(config)
+    CHAT = create(config)
```
Why change the case here?

Purzifal replied:
I use VSCode for coding and the python extension I installed uses pylint, and only changed to stop it showing an error. I do check with flake8 before a commit

connor4312 commented on ChatEventHandler.py:
```diff
@@ -0,0 +1,117 @@
+        if 'whisper' in data['data']['message']['meta']:
+            # format chat messages (whispers)
+            print(data['data']['user_name'] + ' → ' + data['data']['target'],
+                  ': ' + format_msg(data['data']))
```
Consider `string.format` in scenarios such as these

connor4312 commented on ChatEventHandler.py:
```diff
@@ -0,0 +1,117 @@
+CONFIG = configuration
+
+
+def handler(data, chat):
```
Generally it might be better to refactor this into an extensible class or set of classes. This works for fast script hacking, but I'd rather provide a solid foundation that others can build upon in our official example libraries.

Commit:
Made some changes suggested by connor4312

config.example.py expanded on comments providing more details and better
instructions. ChatEventHandler.py Change to class, calls functions from
a dict based on event names, changes string formatting. README.md
changed as per connor's suggestion. chatty\sockets.py fixed an issue
where an error occurs if connection to chat server is lost (my error)
